proposals/NNNN-negative-availability.md

```diff
@@ -0,0 +1,179 @@
+
+```swift
+// if NOT in iOS 13, load the window.
+// Post iOS 13 the window is loaded later in the lifecycle, in the SceneDelegate.
+if #available(iOS 13, *) {} else {
+  loadMainWindow()
+}
+```
+
+Unfortunately, this makes it easy to mistake it for a regular availability check. If the comment isn't clear that the statement is wrong on purpose, problems involving this check could easily land unnoticed in code-reviews. A less noisy way would be to put it under a `guard`:
+
+```swift
+guard #available(iOS 13, *) else {
+  loadMainWindow()
+  return
+}
+// no-op
+```
+
+However, this goes against the code style recommendations involving the usage of `guard`. The guarded part should be the happy path, which is not the case when doing unavailability checks. As shown, it's currently impossible to write an unavailability check that properly fits a developer's engineering expectations and Swift's general style guide. 
+
+Currently, any iOS application that supports UIScenes will face this issue and have to write this workaround. To describe it in a generic way, this issue will be encountered when dealing with any API addition or change that requires more than one availability condition to be implemented.
+
+### Usage of deprecated APIs
+
+A negative availability condition might also be necessary in cases where an API is _informally_ obsoleted; i.e. marked as deprecated and documented as non-functional in newer OS versions.
+
+An example is the deprecation of the [`isAdvertisingTrackingEnabled`](https://developer.apple.com/documentation/adsupport/asidentifiermanager/1614148-isadvertisingtrackingenabled) property. Apps supporting iOS 14 must now use the new App Tracking Transparency framework for user tracking purposes, which involves displaying a permission alert for the user that explains why they are going to be tracked. A developer might determine that this large change in functionality might warrant a complete refactor of this feature, or simply conclude that the negative UX of displaying a new alert is not worth it and that they should remove this feature entirely. In any case however, the property still works when used with older iOS versions:
+
+```swift
+// If NOT on iOS 14, track this action.
+// Post iOS 14, we must ask for permission to do this.
+// The UX impact is not worth it. Let's not do this at all.
+if #available(iOS 14.0) {
+
+} else {
+  oldIos13TrackingLogic(isEnabled: ASIdentifierManager.shared().isAdvertisingTrackingEnabled)
+}
+```
+
+In this specific case, a company that wants to adopt the new privacy practices will require unavailability checks to prevent breaking old versions of the app. In general, this will be the case when dealing with any API that is now deprecated or _informally_ obsoleted.
```

Comment:
```suggestion
In this specific case, a company that wants to adopt the new privacy practices will require unavailability checks to prevent breaking old versions of the app. In general, this will be the case when dealing with any API that is now deprecated.
```